pkg/core/util.go

```diff
@@ -191,3 +196,30 @@ func headerSliceReverse(dest []*block.Header) {
 		dest[i], dest[j] = dest[j], dest[i]
 	}
 }
+
+// CalculateNetworkFee returns network fee for transaction
+func CalculateNetworkFee(script []byte) (util.Fixed8, int) {
+	var (
+		netFee util.Fixed8
+		size   int
+	)
+	if vm.IsSignatureContract(script) {
+		size += 67 + io.GetVarSize(script)
+		netFee = netFee.Add(opcode.GetPrice(opcode.PUSHDATA1, opcode.PUSHDATA1, opcode.PUSHNULL).Add(ecdsaVerifyInteropPrice))
+	} else if n, m, _, ok := vm.ParseMultiSigContract(script); ok {
+		sizeInv := 66 * m
+		size += io.GetVarSize(sizeInv) + sizeInv + io.GetVarSize(script)
+		netFee = netFee.Add(calculateMultisigFee(m)).Add(calculateMultisigFee(n))
+		netFee = netFee.Add(opcode.GetPrice(opcode.PUSHNULL)).Add(util.Fixed8FromInt64(ecdsaVerifyInteropPrice.IntegralValue() * int64(n)))
+	}
```

Comment:
Hmm. `else` what?

Reply:
Actually, that suggests that something is wrong with this check. You're trying to analyze bytecode when you should just run it and see whether it fits in the limit.